src/Tools/dotnet-monitor/CollectionRules/Configuration/CollectionRuleTriggerOptionsProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Configuration
+{
+    internal sealed class CollectionRuleTriggerOptionsProvider :
+        ICollectionRuleTriggerOptionsProvider
+    {
+        private readonly IEnumerable<ICollectionRuleTriggerProvider> _triggerProviders;
+
+        public CollectionRuleTriggerOptionsProvider(IEnumerable<ICollectionRuleTriggerProvider> triggerProviders)
+        {
+            _triggerProviders = triggerProviders;
+        }
+
+        public bool TryGetOptionsType(string triggerType, out Type optionsType)
+        {
+            ICollectionRuleTriggerProvider triggerProvider = _triggerProviders
+                .FirstOrDefault(provider => string.Equals(triggerType, provider.TriggerType, StringComparison.Ordinal));
```

Comment:
is name uniqueness here as well as actiontype enforced at validation time?

Reply:
I think I need to preprocess the list in the ctor, store the information as a dictionary for easier lookup, and log warnings for duplicates. Duplicates is basically not going to happen unless this opens up for extensibility in the future.